cardano_node_tests/tests/test_cli.py

```diff
@@ -290,6 +290,190 @@ def test_address_info_with_invalid_address(self, cluster: clusterlib.ClusterLib,
         assert "Invalid address" in err_str, err_str
 
 
+@pytest.mark.smoke
+class TestAddressBuild:
+    """Tests for cardano-cli address build."""
+
+    @pytest.fixture(scope="class")
+    def stake_address_option_unusable(self) -> bool:
+        return not clusterlib_utils.cli_has("address build --stake-address")
+
+    @allure.link(helpers.get_vcs_link())
+    @pytest.mark.parametrize("payment", ("vkey", "vkey_file", "script_file"))
+    @pytest.mark.parametrize(
+        "stake",
+        (
+            None,
+            "vkey",
+            "vkey_file",
+            "script_file",
+            "address",
+        ),
+    )
+    def test_address_build(
+        self,
+        cluster: clusterlib.ClusterLib,
+        payment: str,
+        stake: str,
+        stake_address_option_unusable: bool,
+    ):
+        """Check `address build` with all valid input options."""
+        if stake == "address" and stake_address_option_unusable:
+            pytest.skip("`stake-address` option is not available on `address build` command")
+
+        temp_template = (
+            f"{common.get_test_id(cluster)}_{payment}_{stake}_{common.unique_time_str()}"
+        )
+
+        payment_vkey_file = DATA_DIR / "golden_payment.vkey"
+
+        stake_vkey_file = DATA_DIR / "golden_stake.vkey"
+        stake_address = "stake_test1uz5mstpskyhpcvaw2enlfk8fa5k335cpd0lfz6chd5c2xpck3nld4"
+
+        script_file = DATA_DIR / "golden_sig.script"
+
+        if payment == "vkey":
+            with open(payment_vkey_file, encoding="utf-8") as infile:
+                # Ignore the first 4 chars, just an informative keyword
+                payment_vkey = helpers.encode_bech32(
+                    prefix="addr_vk", data=json.loads(infile.read().strip()).get("cborHex", "")[4:]
+                )
+
+        if stake == "vkey":
+            with open(stake_vkey_file, encoding="utf-8") as infile:
+                # Ignore the first 4 chars, just an informative keyword
+                stake_vkey = helpers.encode_bech32(
+                    prefix="stake_vk", data=json.loads(infile.read().strip()).get("cborHex", "")[4:]
+                )
+
+        cluster.g_address.gen_payment_addr(
```

Comment:
Please also check the generated address. There's lots of different addresses produced by the parametrized test, so just a check that the generated address is in the set of expected addresses will do.